js/ui/appDisplay.js

```diff
@@ -2382,7 +2401,10 @@ const AppIcon = new Lang.Class({
         return [sourceActor.actor];
     },
 
-    _onNewGtkNotificationSource: function(daemon, source) {
+    _sourceAdded: function(tray, source) {
+        if (!(source instanceof NotificationDaemon.GtkNotificationDaemonAppSource))
+            return;
```

Comment:
I don't know if it makes sense to have this limitation. What was the reason for not allowing Fdo notifications?